jumanji/environments/routing/rware/types.py

```diff
@@ -0,0 +1,137 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING, NamedTuple, Union
+
+if TYPE_CHECKING:  # https://github.com/python/mypy/issues/6239
+    from dataclasses import dataclass
+else:
+    from chex import dataclass
+
+from enum import IntEnum
+
+import chex
+
+# Grid channels
+_SHELVES = 0
+_AGENTS = 1
+
+
+class Action(IntEnum):
+    """An enumeration of possible actions
+    that an agent can take in the warehouse.
+
+    NOOP - represents no operation.
+    FORWARD - move forward.
+    LEFT - turn left.
+    RIGHT - turn right.
+    TOGGLE_LOAD - toggle loading/offloading a shelf.
+    """
+
+    NOOP = 0
+    FORWARD = 1
+    LEFT = 2
+    RIGHT = 3
+    TOGGLE_LOAD = 4
+
+
+class Direction(IntEnum):
+    """An enumeration of possible directions
+    that an agent can take in the warehouse.
+
+    UP - move up.
+    RIGHT - move right.
+    DOWN - move down.
+    LEFT - move left.
+    """
+
+    UP = 0
+    RIGHT = 1
+    DOWN = 2
+    LEFT = 3
+
+
+class Position(NamedTuple):
+    """A class to represent the 2D coordinate position of entities
+
+    x: the x-position of the entity.
+    y: the y-position of the entity.
+    """
+
+    x: chex.Array  # ()
+    y: chex.Array  # ()
```

Comment:
Not really related to this PR, but jumanji in general, we should probably make a general position struct. I know at least connector could've used one also and I reckon most grid worlds would benefit from it. @clement-bonnet thoughts?

Reply:
This is something we could consider later for the whole of Jumanji. Resolving here for now.